tests/drivers/ipm/src/main.c

```diff
@@ -102,6 +104,9 @@ void main(void)
 	 * automation purposes?
 	 */
 
+#ifdef CONFIG_SMP
+	k_sleep(1000 * DELAY_US);
```

Comment:
What is this sleep for?

Reply:
For SMP, maybe console rx_thread haven't done the printout, but main thread will complete it's work, so add some delay to wait rx thread.

Reply:
I'd feel a lot better if this was synchronized with a semaphore instead.

Reply:
yes, but for now ipm structure doesn't provide mechanism to notify printout done. It's not elegant to just add a semphore or atomic variable in two files. And I would like to review code more to do it. And also I run the test about 1000 times with current design, no failure happened.